sunkit_image/radial.py

```diff
@@ -577,3 +583,118 @@ def fnrgf(
+    upsilon : None or int or tuple of size 2, optional
+        A double-sided gamma function applied to the equalized histograms.
+        See Equation (4.15) in the thesis
+        Defaults to 0.35
+    method : str
+        A string describing which method to use for sorting.
+        Options are {"inplace", "numpy", "scipy"}.
+        Defaults to "numpy".
+    vignette: `astropy.units.Quantity`, optional
+        Set pixels above this radius to black.
+        Defaults to None which is no vignette.
+        One suggested value is ``1.5*u.R_sun``.
+    progress: bool, optional
+        Display a progressbar on the main loop.
+        Defaults to True.
+
+    Returns
+    -------
+    `sunpy.map.Map`
+        A SunPy map that has had the RHEF applied to it.
+
+    References
+    ----------
+    * Gilly & Cranmer 2024, in prep.
+
+    * The implementation is highly inspired by this doctoral thesis:
+      Gilly, G. Spectroscopic Analysis and Image Processing of the Optically-Thin Solar Corona
+      https://www.proquest.com/docview/2759080511
+    """
+
+    # Get the radii for every pixel
+    map_r = find_pixel_radii(smap).to(u.R_sun)
+
+    if radial_bin_edges is None:
+        radial_bin_edges = equally_spaced_bins(0, 2, smap.data.shape[0] // 2)
+        radial_bin_edges *= u.R_sun
+
+    # Make sure bins are in the map.
+    if radial_bin_edges[1, -1] > np.max(map_r):
+        radial_bin_edges = equally_spaced_bins(
+            inner_value=radial_bin_edges[0, 0],
+            outer_value=np.max(map_r),
+            nbins=radial_bin_edges.shape[1],
+        )
+
+    # Select the sort method
+    if method == "inplace":
+        the_func = percentile_ranks_numpy_inplace
+    elif method == "numpy":
+        the_func = percentile_ranks_numpy
+    elif method == "scipy":
+        the_func = percentile_ranks_scipy
+    else:
+        msg = f"{method} is invalid. Allowed values are 'inplace', 'numpy', 'scipy'"
+        raise NotImplementedError(msg)
+
+    # Allocate storage for the filtered data
+    data = np.zeros_like(smap.data)
+    meta = smap.meta
+    # Calculate the filter values for each radial bin.
+    if progress is None:
+        progress = True
+    for i in tqdm(range(radial_bin_edges.shape[1]), desc="RHEF: ", disable=not progress):
+        # Identify the appropriate radial slice
+        here = np.logical_and(map_r >= radial_bin_edges[0, i], map_r < radial_bin_edges[1, i])
+        if application_radius is not None and application_radius > 0:
+            here = np.logical_and(here, map_r >= application_radius)
+
+        # Perform the filtering operation
+        data[here] = the_func(smap.data[here])
+        if upsilon is not None:
+            data[here] = apply_upsilon(data[here], upsilon)
+    new_map = sunpy.map.Map(data, meta)
+
+    if vignette is not None:
+        new_map = blackout_pixels_above_radius(new_map, vignette)
+    else:
+        new_map = blackout_pixels_above_radius(new_map, 1.5 * u.R_sun)
+
```

Comment:
```suggestion
    new_map.plot_settings["norm"] = None
```
Hopefully that will stop us needed to do norm when we plot the new maps.